### PR TITLE
[Feature] Config management subcommand

### PR DESCRIPTION
**Pull request by kentwelcome**
Signed-off-by: Kent Huang <kent@infuseai.io>

**PR checklist**

- [x] Ensure you have added or ran the appropriate tests for your PR.
- [x] DCO signed

**What type of PR is this?**
Enhance UX

**What this PR does / why we need it**:
https://www.notion.so/infuseai/PipeRider-CLI-Config-Management-eebb112ff55e4e4ebbe24115fb014173
 - `piperider config list-datasource`   : List current config
 - `piperider config add-datasource`    : Add new datasource into current workspace
 - `piperider config delete-datasource` : Delete datasource from current workspcae

**Which issue(s) this PR fixes**:
n/a

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

### CODE REVIEW

File: piperider_cli/cli.py [hunk 1]
```diff
@@ -232,6 +233,61 @@ def compare_reports(**kwargs):
                        debug=kwargs.get('debug', False))
 
 
+@cli.group('config', help='Manage the Pipeline configurations.', cls=BetaGroup)
```
Comment (wcchang1115) [inline piperider_cli/cli.py:236]: is it a typo? Pipeline -> PipeRider

File: piperider_cli/configuration.py [hunk 2]
```diff
@@ -285,6 +285,11 @@ def ask_for_datasource(self):
             answers = inquirer.prompt(questions, raise_keyboard_interrupt=True)
             return answers['datasource']
 
+    def delete_datasource(self, datasource):
+        if datasource in self.dataSources:
+            self.dataSources.remove(datasource)
+            self.dump(PIPERIDER_CONFIG_PATH)
```
Comment (wcchang1115) [inline piperider_cli/configuration.py:291]: due to `dump` implementation, users' change of `profiler`, `table inclusion/exclusion`, and `tables` in `config.yml` would be overwritten by config template

File: piperider_cli/initializer.py [hunk 3]
```diff
@@ -110,3 +115,100 @@ def exec(dbt_project_path=None, dbt_profiles_dir=None):
...
+    def add(report_dir=None):
+        console = Console()
+        config = Configuration.load()
+        console.rule('Add datasource')
+        cls, name = DataSource.ask()
+        ds: DataSource = cls(name=name)
+        config.dataSources.append(ds)
+        if _ask_user_update_credentials(ds):
+            _generate_piperider_workspace()
+            config.dump(PIPERIDER_CONFIG_PATH)
```
Comment (wcchang1115) [inline piperider_cli/initializer.py:210]: the same issue as above due to `config.dump` implementation

File: piperider_cli/event/__init__.py [hunk 4]
```diff
@@ -127,3 +129,12 @@ def log_event(prop, event_type, **kwargs):
         **prop,
     )
     _collector.log_event(payload, event_type)
+
+
+class UserProfileConnector(object):
```
Comment (wcchang1115) [inline piperider_cli/event/__init__.py:134]: just a minor question, why do we name it `Connector`?

File: piperider_cli/event/__init__.py [hunk 5]
```diff
@@ -127,3 +129,12 @@ def log_event(prop, event_type, **kwargs):
...
+    @staticmethod
+    def update(key: str, val: Any, name: str = None):
```
Comment (wcchang1115) [inline piperider_cli/event/__init__.py:136]: could we allow the current `val` type annotation to be `Union[str, bool]`?

**Review by wcchang1115 — CHANGES_REQUESTED** [review]
The major change request is when dumping `config.yml`, the users' change would be overwritten.
Thanks!

File: piperider_cli/cli.py (hunk 1 above)
Comment (kentwelcome) [reply piperider_cli/cli.py]: fixed

File: piperider_cli/configuration.py (hunk 2 above)
Comment (kentwelcome) [reply piperider_cli/configuration.py]: Implement a new method of `Configuration` class called `flush_datasource`. This method will only update the `dataSources` field in `.piperider/config.yml`. It won't overwrite the other part of `config.yml` Even the user comments in `config.yml` will be conserved.

File: piperider_cli/initializer.py (hunk 3 above)
Comment (kentwelcome) [reply piperider_cli/initializer.py]: change to use method `flush_datasource` instead of `dump`

File: piperider_cli/event/__init__.py (hunk 5 above)
Comment (kentwelcome) [reply piperider_cli/event/__init__.py]: fixed

File: piperider_cli/event/__init__.py (hunk 4 above)
Comment (kentwelcome) [reply piperider_cli/event/__init__.py]: Rename it to `UserProfileCongigurator`

**Comment by kentwelcome** [comment]
> The major change request is when dumping `config.yml`, the users' change would be overwritten. Thanks!

Implement a new method of Configuration class called flush_datasource. This method will only update the dataSources field in .piperider/config.yml. It won't overwrite the other part of config.yml Even the user comments in config.yml will be conserved.

**Review by popcornylu — CHANGES_REQUESTED** [review]
If the data source exists, should show error message.

**Comment by kentwelcome** [comment]
> If the data source exists, should show error message.

Will block the user fill in the duplicate data source name when we are asking the data source name.

**Review by popcornylu — APPROVED** [review]

**Review by popcornylu — APPROVED** [review]
LGTM

**Pull request merged**